src/CombFilter/CombFilter.cpp

```diff
@@ -0,0 +1,145 @@
+//
+//  CombFilter.cpp
+//  CombFilter
+//
+//  Created by Vedant Kalbag on 09/02/22.
+//
+#include "ErrorDef.h"
+#include "CombFilter.h"
+
+
+// =====================================================================================================
+// CCombFilterBase method definitions
+CCombFilterBase::CCombFilterBase(int delayLength, int iNumChannels, float gainValue)
+{
+    m_iNumChannels = iNumChannels;
+    m_fGainValue = gainValue;
+    m_iDelayValueSamples = delayLength;
+    m_RingBuffer = new CRingBuffer<float>* [m_iNumChannels];
+    for (int i=0;i<m_iNumChannels;i++)
+    {
+        m_RingBuffer[i] = new CRingBuffer<float>(m_iDelayValueSamples);
+        m_RingBuffer[i]->reset();
+    }
+}
+//CCombFilterBase::~CCombFilterBase()
+//{
+//}
+
+//Error_t CCombFilterBase::init(int iNumChannels, float delayValue, float gainValue)
+//{
+//
+//}
+
+Error_t CCombFilterBase::setGain(float fGainValue)
+{
+    //Set gainValue to fGainValue
+    m_fGainValue = fGainValue;
+    return Error_t::kNoError;
+}
+
+float CCombFilterBase::getGain()
+{
+    return m_fGainValue;
+}
+
+Error_t CCombFilterBase::setDelay(int iDelayValueSamples)
+{
+    //Set delayValue to iDelayValue
+    m_iDelayValueSamples = iDelayValueSamples;
```

Comment:
I'm a bit confused as to where this actually gets used. `m_iDelayValueSamples` is used in the constructor when creating the ring buffers (which determines the max delay), but this method should change the actual delay (i.e. modify the read index), and I can't find where that happens.
(I noticed this because I was surprised that it always returns `kNoError`; what if the user tries to set a delay which is greater than the max delay (or less than 0)?)